openedx/core/djangoapps/user_api/views.py

```diff
@@ -229,6 +241,25 @@ def get(self, request):
         for field_name in self.DEFAULT_FIELDS:
             self.field_handlers[field_name](form_desc, required=True)
 
+        # Custom form fields can be added via the form set in settings.REGISTRATION_EXTENSION_FORM
+        custom_form = get_custom_form()
+
+        for field_name, field in custom_form.fields.items():
+            restrictions = {}
+            if getattr(field, 'max_length', None):
+                restrictions['max_length'] = field.max_length
+            if getattr(field, 'min_length', None):
+                restrictions['min_length'] = field.min_length
+            form_desc.add_field(
+                field_name, label=field.label,
+                default=self.field_options(custom_form, field_name).get('default'),
+                field_type=self.field_type(custom_form, field_name, field),
+                placeholder=field.initial, instructions=field.help_text, required=field.required,
```

Comment:
Nit: this would be easier to read if each `kwarg=value` was on a separate line, rather than some of them sharing lines.